Content.Server/_EE/Xelthia/XelthiaSystem.cs

```diff
@@ -0,0 +1,116 @@
+using Content.Shared.Body.Part;
+using Robust.Shared.Audio;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Placement;
+using Robust.Shared.Random;
+using Robust.Shared.Serialization.Manager;
+using Robust.Shared.Timing;
+using Content.Shared.Chemistry.Components;
+
+
+namespace Content.Server._EE.Xelthia;
+
+
+/// <summary>
+/// Xelthia Arm Regrowth stuff
+/// </summary>
+public sealed class XelthiaSystem : EntitySystem
+{
+    /// <inheritdoc/>
+
+    [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+    [Dependency] private readonly IEntityManager _entityManager = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly BloodstreamSystem _bloodstreamSystem = default!;
+
+    //public const string XelthiaRegenerateActionId = "XelthiaRegenerateAction";
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<XelthiaComponent, ComponentStartup>(OnInit);
+        SubscribeLocalEvent<XelthiaComponent, ArmRegrowthEvent>(OnRegrowthAction);
+    }
+
+    private void OnInit(EntityUid uid, XelthiaComponent component, ComponentStartup args)
+    {
+        _actionsSystem.AddAction(uid, ref component.XelthiaRegenerateAction, out var regenerateAction, "XelthiaRegenerateAction");
+        if (regenerateAction?.UseDelay != null)
+            component.UseDelay = regenerateAction.UseDelay.Value;
+    }
+
+    private void OnRegrowthAction(EntityUid uid, XelthiaComponent component, ArmRegrowthEvent args)
+    {
+        var start = _gameTiming.CurTime;
+        var end = _gameTiming.CurTime + component.UseDelay;
+        _actionsSystem.SetCooldown(component.XelthiaRegenerateAction!.Value, start, end); // Cooldown
+
+        //IEntityManager entityManager = uid;
+        IEntityManager entityManager = base.EntityManager; // There's no way this is good code. I don't know what im doing though, so. I dunno.
+        var bodySystem = entityManager.System<BodySystem>();
+        var transformSystem = entityManager.System<SharedTransformSystem>();
+
+        if (!entityManager.TryGetComponent(uid, out BodyComponent? body)
+            || !entityManager.TryGetComponent(uid, out TransformComponent? xform))
+            return;
+
+        var root = bodySystem.GetRootPartOrNull(uid, body);
+        if (root is null)
+            return;
+
+        var parts = bodySystem.GetBodyChildrenOfType(uid, BodyPartType.Arm, body); // Deletes Arms and hands
+        foreach (var part in parts)
+        {
+
+            foreach (var child in bodySystem.GetBodyPartChildren(part.Id, part.Component))
+                entityManager.QueueDeleteEntity(child.Id);
+
+            transformSystem.AttachToGridOrMap(part.Id);
+            entityManager.QueueDeleteEntity(part.Id);
+            entityManager.SpawnAtPosition("FoodMeatXelthiaTentacle", xform.Coordinates); // Should drop arms equal to the number attached
```

Comment:
```suggestion
        var bodySystem = _entityManager.System<BodySystem>();
        var transformSystem = _entityManager.System<SharedTransformSystem>();

        if (!_entityManager.TryGetComponent(uid, out BodyComponent? body)
            || !_entityManager.TryGetComponent(uid, out TransformComponent? xform))
            return;

        var root = bodySystem.GetRootPartOrNull(uid, body);
        if (root is null)
            return;

        var parts = bodySystem.GetBodyChildrenOfType(uid, BodyPartType.Arm, body); // Deletes Arms and hands
        foreach (var part in parts)
        {

            foreach (var child in bodySystem.GetBodyPartChildren(part.Id, part.Component))
                _entityManager.QueueDeleteEntity(child.Id);

            transformSystem.AttachToGridOrMap(part.Id);
            _entityManager.QueueDeleteEntity(part.Id);
            _entityManager.SpawnAtPosition("FoodMeatXelthiaTentacle", xform.Coordinates); // Should drop arms equal to the number attached
```
Test this before you apply it!